### PR TITLE
DOC accessing named colors

### PR DESCRIPTION
**Pull request by NelleV**
The example now displays all named colors.

### CODE REVIEW

**Comment by anntzer** [comment]
Similarly to https://github.com/matplotlib/matplotlib/pull/9530#issuecomment-338505824 I think it'd be nice to have a link from http://matplotlib.org/api/colors_api.html to this example.

File: examples/color/named_colors.py [hunk 1]
```diff
@@ -1,55 +1,98 @@
...
-    y = Y - (row * h) - h
 
-    xi_line = w * (col + 0.05)
-    xf_line = w * (col + 0.25)
-    xi_text = w * (col + 0.3)
+###############################################################################
+# Base colors
+# ===========
+#
+# Available in Matplotlib since the age of dawn, the base colors offer the
```
Comment (anntzer) [inline examples/color/named_colors.py]: comments and actual colors are inverted between base colors and css4 colors.

File: examples/color/named_colors.py [hunk 2]
```diff
@@ -1,55 +1,98 @@
...
-colors = dict(mcolors.BASE_COLORS, **mcolors.CSS4_COLORS)
+def plot_colors(colors):
+    # Sort colors by hue, saturation, value and name.
+    by_hsv = sorted(
+        (tuple(mcolors.rgb_to_hsv(mcolors.to_rgba(color)[:3])), name)
+        for name, color in colors.items())
+    sorted_names = [name for hsv, name in by_hsv]
+
+    n = len(sorted_names)
+    if n > 400:
```
Comment (anntzer) [inline examples/color/named_colors.py]: I would just stick to 4 columns everywhere, I don't think the overly wide figure at the bottom of https://2855-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/color/named_colors.html#sphx-glr-gallery-color-named-colors-py is really an improvement.

File: examples/color/named_colors.py [hunk 3]
```diff
@@ -1,55 +1,98 @@
...
-            horizontalalignment='left',
-            verticalalignment='center')
 
-    ax.hlines(y + h * 0.1, xi_line, xf_line,
-              color=colors[name], linewidth=(h * 0.6))
+###############################################################################
+# Tableau colors
+# ==============
+#
+# The tableau colors are the default the default in Matplotlib since 2.0, and
```
Comment (QuLogic) [inline examples/color/named_colors.py]: Duplicate "the default"

File: examples/color/named_colors.py (hunk 2 above)
Comment (tacaswell) [reply examples/color/named_colors.py]: I think I agree, it spans off my wide screen.

**Pull request closed without merging**